DFC.ServiceTaxonomy.GraphSync/Services/SynonymService.cs

```diff
@@ -0,0 +1,42 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using DFC.ServiceTaxonomy.GraphSync.Queries;
+using DFC.ServiceTaxonomy.GraphSync.Services.Interface;
+using DFC.ServiceTaxonomy.Neo4j.Services;
+
+namespace DFC.ServiceTaxonomy.GraphSync.Services
+{
+    public class SynonymService : ISynonymService
+    {
+        private readonly IGraphDatabase _neoGraphDatabase;
+
+        public SynonymService(IGraphDatabase neoGraphDatabase)
+        {
+            _neoGraphDatabase = neoGraphDatabase ?? throw new ArgumentNullException(nameof(neoGraphDatabase));
+        }
+
+        public IEnumerable<string> GetSynonyms(string node)
+        {
+            var query = new MatchSynonymsQuery();
```

Comment:
I don't think we need to create an empty `MatchSynonymsQuery`, which gets thrown away if node is skill or occupation.

Then we could remove the constructor of `MatchSynonymsQuery` with no parameters and there would be no need to call `this.CheckIsValid()` in the query (which it should do, if there's a possibility that the params haven't been supplied).

If node is not skill or occupation we should properly handle it, rather than execute a query with empty string params.

Reply:
Yeah valid point, I was trying to initialise the query and set it before running it.

A simple refactor would remove this need entirely, updating now